crates/uv-python/src/managed.rs

```diff
@@ -323,6 +323,36 @@ impl ManagedPythonInstallation {
         }
     }
 
+    /// Ensure the environment contains the canonical Python executable names.
+    pub fn ensure_canonical_executables(&self) -> Result<(), Error> {
+        let python = self.executable();
+
+        // Workaround for python-build-standalone v20241016 which is missing the standard Python
+        // executable on Windows.
+        // See https://github.com/astral-sh/uv/issues/8298
+        if !python.try_exists()? {
+            match self.key.variant {
+                PythonVariant::Default => {}
+                PythonVariant::Freethreaded => {
+                    let existing = self.python_dir().join(format!(
+                        "python{}.{}t{}",
+                        self.key.major,
+                        self.key.minor,
+                        std::env::consts::EXE_SUFFIX
+                    ));
```

Comment:
What if this executable doesn't exist? Should we explicitly test that and error if not?

I assume the symlink/copy below will fail, but will it fail with a comprehensible error?

Reply:
Yeah I thought about testing it but was okay with the failure. It should "never occur" and we should get some error context but we could probably add more. I'll try to do that.

Reply:
We have a couple error variants for this now.